agent/lib/kontena/service_pods/creator.rb

```diff
@@ -24,7 +24,9 @@ def perform
         end
         service_container = get_container(service_pod.name)
 
-        sleep 1 until Celluloid::Actor[:network_adapter].running?
+        if !Celluloid::Actor[:network_adapter].wait_running? { debug "waiting for weave to be running..." }
+          raise "Timeout waiting for weave to be running"
```

Comment:
Hmm... what should the agent do in this case?

Seems like the CLI just spins in deploying... if the agent gives up.